src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1157,17 +1205,13 @@ private void WriteReferenceDocumentationFiles()
 
                         foreach (string fileExtension in relatedExtensions.Split(RelatedPropertySeparator))
                         {
-                            if (fileExtension.ToLower() == extension)
+                            if (fileExtension.ToLowerInvariant() == extension)
```

Comment:
Should be `StringComparer.InvariantCultureIgnoreCase.Equals(fileExtension, extension)` to avoid allocations.